frontend/app/src/components/StatusWidget/StatusWidget.tsx

```diff
@@ -136,237 +106,244 @@ const RUNNING_MAN_DISPLAY_DELAY_TIME_MS = 500
  * Displays various script- and connection-related info: our WebSocket
  * connection status, the run-state of our script, and other transient events.
  */
-class StatusWidget extends PureComponent<StatusWidgetProps, State> {
-  /** onSessionEvent signal connection */
-  private sessionEventConn?: SignalConnection
-
-  private curView?: ReactNode
-
-  private readonly minimizePromptTimer = new Timer()
+const StatusWidget: React.FC<StatusWidgetProps> = ({
+  connectionState,
+  sessionEventDispatcher,
+  scriptRunState,
+  rerunScript,
+  stopScript,
+  allowRunOnSave,
+  //theme,
+}): ReactElement => {
```

Comment:
**nit:** Because this uses `React.FC`, the explicit return type here isn't necessary

Reply:
[caf4405](https://github.com/streamlit/streamlit/pull/9851/commits/caf4405c3566f7f913a6ac56a1eb47ea7d963a0f)